File: 8300. Revisão de contrato/contrato.c
#include <stdio.h>
#include <string.h>

int main(){
	char numeros[100];
	char digito;
	int i;
	int mostrei;

	scanf("%c", &digito);
	while(digito != '0'){
		scanf("%s", numeros);
		mostrei = 0;
		for(i = 0; i < strlen(numeros); i++){
			if(numeros[i] != digito && (numeros[i] != '0' || mostrei)
				){
				printf("%c", numeros[i]);
				mostrei++;
			}
		}
		if(mostrei == 0){
			printf("0");
		}
		printf("\n");

		scanf("%*c");
		scanf("%c", &digito);
	}


	return 0;
}

File: 8300. Revisão de contrato/CONTRATO.java
import java.util.Scanner;


class CONTRATO{
	public static void main(String[] args){
		int digito;
		String numeros;

		Scanner entrada = new Scanner(System.in);

		digito = entrada.nextInt();
		
		while(digito != 0){
			numeros = entrada.nextLine().trim();

			numeros = numeros.replace(Integer.toString(digito), "");
			if(numeros.isEmpty()){
				numeros = "0";
			}else{
				int inicio = 0;
				for(int i = 0; i < numeros.length() - 1; i++){
					if(numeros.startsWith("0")){
						inicio++;
					}else{
						break;
					}
				}
				numeros = numeros.substring(inicio);
			}
			System.out.println(numeros);

			digito = entrada.nextInt();
		}
	}
}
